sharding/contracts/sharding_manager_test.go

```diff
@@ -16,412 +16,320 @@ import (
+// TestNotaryRegister tests notary registers in a normal condition.
+func TestNotaryRegister(t *testing.T) {
+	// Initializes 3 accounts to register as notaries.
+	const notaryCount = 3
+	notaryPool, genesis := initAccounts(notaryCount)
+	backend := backends.NewSimulatedBackend(genesis)
+	_, _, smc, _ := deploySMCContract(backend, notaryPool[0].privKey)
 
+	// Verify notary 0 has not registered.
+	notary, err := smc.NotaryRegistry(&bind.CallOpts{}, notaryPool[0].addr)
+	if err != nil {
+		t.Fatalf("Can't get notary registry info: %v", err)
+	}
 	if notary.Deposited {
-		t.Errorf("Notary deposited with insufficient fund")
+		t.Errorf("Notary has not registered. Got deposited flag: %v", notary.Deposited)
 	}
 
-	if numNotaries.Cmp(big.NewInt(0)) != 0 {
-		t.Errorf("Incorrect count from notary pool. Want: 0, Got: %v", numNotaries)
-	}
+	// Test notary 0 has registered.
+	notaryPool.registerNotaries(t, backend, smc, notaryDeposit, 0, 1)
+	// Test notary 1 and 2 have registered.
+	notaryPool.registerNotaries(t, backend, smc, notaryDeposit, 1, 3)
+	// Check total numbers of notaries in pool, should be 3
+	checkNotaryPoolLength(t, backend, smc, big.NewInt(3))
+}
 
+// TestNotaryRegisterInsufficientEther tests notary registers with insufficient deposit.
+func TestNotaryRegisterInsufficientEther(t *testing.T) {
+	notaryPool, genesis := initAccounts(1)
+	backend := backends.NewSimulatedBackend(genesis)
+	_, _, smc, _ := deploySMCContract(backend, notaryPool[0].privKey)
+	if err := notaryPool.registerNotaries(t, backend, smc, notaryDepositInsufficient, 0, 1); err == nil {
+		t.Errorf("Notary register should have failed with insufficient deposit")
+	}
+	checkNotaryPoolLength(t, backend, smc, big.NewInt(0))
 }
 
 // TestNotaryDoubleRegisters tests notary registers twice.
 func TestNotaryDoubleRegisters(t *testing.T) {
-	addr := crypto.PubkeyToAddress(mainKey.PublicKey)
-	backend := backends.NewSimulatedBackend(core.GenesisAlloc{addr: {Balance: accountBalance2000Eth}})
-	txOpts := bind.NewKeyedTransactor(mainKey)
-	txOpts.Value = notaryDeposit
-	_, _, smc, _ := deploySMCContract(backend, mainKey)
+	notaryPool, genesis := initAccounts(1)
+	backend := backends.NewSimulatedBackend(genesis)
+	_, _, smc, _ := deploySMCContract(backend, notaryPool[0].privKey)
 
 	// Notary 0 registers.
-	smc.RegisterNotary(txOpts)
-	backend.Commit()
-
-	notary, _ := smc.NotaryRegistry(&bind.CallOpts{}, addr)
-	numNotaries, _ := smc.NotaryPoolLength(&bind.CallOpts{})
-
-	if !notary.Deposited {
-		t.Errorf("Notary has not registered. Got deposited flag: %v", notary.Deposited)
-	}
-
-	if numNotaries.Cmp(big.NewInt(1)) != 0 {
-		t.Errorf("Incorrect count from notary pool. Want: 1, Got: %v", numNotaries)
-	}
+	notaryPool.registerNotaries(t, backend, smc, notaryDeposit, 0, 1)
+	checkNotaryPoolLength(t, backend, smc, big.NewInt(1))
 
-	// Notary 0 registers again.
-	_, err := smc.RegisterNotary(txOpts)
-	if err == nil {
+	// Notary 0 registers again. This time should fail
```

Comment:
comment formatting, maybe we can find a way to automatically do this if we accidentally miss a period? Can travis do this?